src/TimeSpans.jl

```diff
@@ -124,6 +149,16 @@ function translate(span, by::Period)
     return TimeSpan(start(span) + by, stop(span) + by)
 end
 
+"""
+    extend(span, by::Period)
+
+Returns `TimeSpan(start(span), max(start(span), stop(span) + by)`
+"""
+function extend(x, by::Period)
+    newstop = convert(Nanosecond, stop(x)) + convert(Nanosecond, by)
+    return TimeSpan(start(x), max(start(x), newstop))
+end
```

Comment:
Should fail in this case:
```julia
julia> extend(TimeSpan(Nanosecond(0), Nanosecond(1)), -Nanosecond(5))
TimeSpan(00:00:00.000000000, 00:00:00.000000001)
```

Reply:
I'll mention an alternative solution to this is to allow for `start > stop` but in such cases you're making a null timespan